singularity-opac/neutrinos/mean_neutrino_s_variant.hpp

```diff
@@ -0,0 +1,104 @@
+// ======================================================================
+// © 2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef SINGULARITY_OPAC_NEUTRINOS_MEAN_NEUTRINO_S_VARIANT_
+#define SINGULARITY_OPAC_NEUTRINOS_MEAN_NEUTRINO_S_VARIANT_
+
+#include <utility>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-opac/base/opac_error.hpp>
+#include <singularity-opac/base/radiation_types.hpp>
+#include <singularity-opac/neutrinos/neutrino_s_variant.hpp>
+#include <variant/include/mpark/variant.hpp>
+
+namespace singularity {
+namespace neutrinos {
+namespace impl {
+
+template <typename... SOpacs>
+class MeanSVariant {
```

Comment:
Variant here is for units vs no units?

Reply:
Yes. Supposing I trade the variant over units vs no units (like this one) for a template parameter, how will that interact with parthenon-like params? For example in tasks I will want to call
```c++
auto &mean_opac = params->Get<MeanOpacity>("mean_opacity");
```
but if my type becomes `MeanOpacity<WhetherIHaveUnits>` and `WhetherIHaveUnits` is selectable at runtime when `mean_opac` is enrolled in the `Params` class, how do I pull it back out in the code later without also templating my task on `WhetherIHaveUnits` as well?

Reply:
I was imagining that in Phoebus, we just always have units. And non-cgs units just sets the conversion factor to 1? Is there a reason that wouldn't work? We could even do something like
```C++
typedef MeanS = MeanSWithUnits<MeanSGray>;
```
or something silly like that

Reply:
Hm I'd have to think about this a bit more... Let me open an issue since I've already got a lot of different contexts loaded into my brain's RAM right now: #35